plugins/woocommerce-admin/client/marketplace/components/product-list-content/no-results.tsx

```diff
@@ -1,159 +1,152 @@
 /**
  * External dependencies
  */
-import { __, sprintf } from '@wordpress/i18n';
+import { __ } from '@wordpress/i18n';
 import { useEffect, useState } from '@wordpress/element';
 import { useQuery } from '@woocommerce/navigation';
 
 /**
  * Internal dependencies
  */
-import NoResultsExtensionsIcon from '../../assets/images/no-results-extensions.svg';
-import NoResultsThemesIcon from '../../assets/images/no-results-themes.svg';
 import { fetchDiscoverPageData, ProductGroup } from '../../utils/functions';
 import ProductLoader from '../product-loader/product-loader';
 import ProductList from '../product-list/product-list';
-import './no-results.scss';
-import { ProductType } from '../product-list/types';
+import { ProductType, SearchResultType } from '../product-list/types';
 import CategorySelector from '../category-selector/category-selector';
+import './no-results.scss';
 
-interface NoResultsProps {
-	type: ProductType;
-}
-
-export default function NoResults( props: NoResultsProps ): JSX.Element {
-	const [ productGroup, setProductGroup ] = useState< ProductGroup >();
-	const [ isLoadingProductGroup, setisLoadingProductGroup ] =
-		useState( false );
-	const [ noResultsTerm, setNoResultsTerm ] = useState< string >( '' );
-	const typeLabel =
-		props.type === ProductType.theme ? 'themes' : 'extensions';
+export default function NoResults( props: {
+	type: SearchResultType;
+	showHeading?: boolean;
+	heading?: string;
+} ): JSX.Element {
+	const [ productGroups, setProductGroups ] = useState< ProductGroup[] >();
+	const [ isLoading, setIsLoading ] = useState( false );
 	const query = useQuery();
 	const showCategorySelector = query.tab === 'search' && query.section;
+	const productGroupsForSearchType: {
+		[ key in SearchResultType ]: string[];
+	} = {
+		[ SearchResultType.all ]: [ 'most-popular', 'popular-themes' ],
+		[ SearchResultType.theme ]: [ 'popular-themes' ],
+		[ SearchResultType.extension ]: [ 'most-popular' ],
+	};
```

Comment:
TypeScript linter in my IDE complains about the type here:

```
Initializer type {} is not assignable to variable type {[key in SearchResultType]: string[]} 
```

It suggests two alternatives which make it happy. The first is:

```ts
	const productGroupsForSearchType: {} = {
		[ SearchResultType.all ]: [ 'most-popular', 'popular-themes' ],
		[ SearchResultType.theme ]: [ 'popular-themes' ],
		[ SearchResultType.extension ]: [ 'most-popular' ]
	};
```

The second is this:

```ts
	const productGroupsForSearchType: {
		[ key in SearchResultType ]: string[];
	} = {
		all: [], extension: [], theme: [],
		[ SearchResultType.all ]: [ 'most-popular', 'popular-themes' ],
		[ SearchResultType.theme ]: [ 'popular-themes' ],
		[ SearchResultType.extension ]: [ 'most-popular' ]
	};
```

I'm not sure why it doesn't like your type definition – maybe it'll make more sense to you. I guess the first solution makes the type less well defined.

Reply:
That is odd. I'm not getting anything on VSCode or on PHPStorm. I'll remove the explicit type definition. I won't change much.

```suggestion
	const productGroupsForSearchType = {
		[ SearchResultType.all ]: [ 'most-popular', 'popular-themes' ],
		[ SearchResultType.theme ]: [ 'popular-themes' ],
		[ SearchResultType.extension ]: [ 'most-popular' ],
	};
```